ui/src/app/views/workflow/run/node/pipeline/spawninfo/spawninfo.component.ts

```diff
@@ -125,16 +139,19 @@ export class WorkflowRunJobSpawnInfoComponent implements OnDestroy {
             this.workerSubscription = this.worker.response().subscribe(msg => {
                 if (msg) {
                     let serviceSpawnInfos: Array<SpawnInfo> = JSON.parse(msg);
-                    if (serviceSpawnInfos && serviceSpawnInfos.length > 0) {
-                        this.spawnInfos = this.getSpawnInfos(serviceSpawnInfos);
-                    }
-                    if (this.jobStatus === PipelineStatus.SUCCESS || this.jobStatus === PipelineStatus.FAIL ||
-                        this.jobStatus === PipelineStatus.STOPPED) {
-                        this.stopWorker();
-                        if (this.nodeJobRun.spawninfos && this.nodeJobRun.spawninfos.length > 0) {
-                            this.spawnInfos = this.getSpawnInfos(this.nodeJobRun.spawninfos);
+                    this.zone.run(() => {
+                        this._cd.markForCheck();
```

Comment:
should not be at the bottom ?

Reply:
I don’t think it matters, the current js thread has to end before that angular trigger change detection no?

Reply:
no idea

Reply:
ok so at the end